app/core/voyagerUtils.js

```diff
@@ -0,0 +1,116 @@
+
+module.exports = {
+
+  /**
+   * Creates a level status map using the list of level session objects.
+   * @param {Object[]} sessions - The list of level session objects.
+   * @returns {Object} - Object with key as the level original id, and value as complete/started.
+   */
+  getLevelStatusMap: function (sessions) {
+    const levelStatusMap = {}
+    for (const session of sessions) {
+      const levelOriginal = (session.level || session.get('level') || {}).original
+      if (levelOriginal && levelStatusMap[levelOriginal] !== 'complete') { // Don't overwrite a complete session with an incomplete one
+        if ((session.state || session.get('state') || {}).complete) {
+          levelStatusMap[levelOriginal] = 'complete'
+        } else {
+          levelStatusMap[levelOriginal] = 'started'
+        }
+      }
+    }
+    return levelStatusMap
+  },
+
+  /**
+   * Creates a level data map using the list of level objects.
+   * @param {Object[]} levels - The list of level objects.
+   * @returns {Object} - Object with key as the level original id, and value as the object containing level's data.
+   */
+  getLevelDataMap: function (levels) {
+    const levelDataMap = {}
```

Comment:
Regarding the name of the method, is there a rationale for returning an Object literal vs a [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)?

Should the method be named something like `getLevelData` to not suggest the Map data structure?

Reply:
No particular reason, it's just using an object as a map.
However, I did re-check the usage of this method and it is being used to convert an array of levels to an object, in case its not already an object (https://github.com/codecombat/codecombat/pull/5281/files#diff-27dfda484676dea165b5a8540d8ec1d3R82). 
Returning an object from here will keep it in sync with the case when it is already an object.